app/src/split/bluetooth/central.c

```diff
@@ -265,6 +270,105 @@ static uint8_t split_central_notify_func(struct bt_conn *conn,
     return BT_GATT_ITER_CONTINUE;
 }
 
+#if IS_ENABLED(CONFIG_ZMK_SPLIT_BLE_CENTRAL_BATTERY_LEVEL_FETCHING)
+
+static uint8_t peripheral_battery_levels[ZMK_SPLIT_BLE_PERIPHERAL_COUNT] = {0};
+
+int zmk_split_get_peripheral_battery_level(uint8_t source, uint8_t *level) {
+    if (source >= ARRAY_SIZE(peripheral_battery_levels)) {
+        return -EINVAL;
+    }
+
+    *level = peripheral_battery_levels[source];
+    return 0;
+}
+
+K_MSGQ_DEFINE(peripheral_batt_lvl_msgq, sizeof(struct zmk_peripheral_battery_state_changed),
+              CONFIG_ZMK_SPLIT_BLE_CENTRAL_BATTERY_LEVEL_QUEUE_SIZE, 4);
+
+void peripheral_batt_lvl_change_callback(struct k_work *work) {
+    struct zmk_peripheral_battery_state_changed ev;
+    while (k_msgq_get(&peripheral_batt_lvl_msgq, &ev, K_NO_WAIT) == 0) {
+        LOG_DBG("Triggering peripheral battery level change %u", ev.state_of_charge);
+        peripheral_battery_levels[ev.source] = ev.state_of_charge;
```

Comment:
Should the `peripheral_battery_levels` value also be reset to 0 (or a sentinel value?) when the peripheral disconnects?

Additionally, should there be a matching API to permit consumers to ask about the peripheral's presence without subscribing to `zmk_split_peripheral_status_changed` and maintaining an independent local view on the state?

A consumer of the peripheral battery level such as a display would almost certainly want to display something different when the peripheral is disconnected.

Reply:
Worth noting that https://github.com/zmkfirmware/zmk/pull/2033 adds an API for checking peripheral bonded/connected status.

Reply:
Oh, excellent. Well, then those would play nicely, then. 👍 

In addition to that though, I think it might still still be worth changing `zmk_split_get_peripheral_battery_level` here to return an error code if the specified peripheral is not connected.

Reply:
> Oh, excellent. Well, those would play nicely, then. 👍
> 
> In addition to that though, I think it might still still be worth changing `zmk_split_get_peripheral_battery_level` here to return an error code if the specified peripheral is not connected.

Good call. Added that check and the API now returns `-ENOTCONN` in that scenario. Pushed the changes.

Reply:
Ok, I've also added code to trigger an event on disconnect to set the level back to 0 for the newly disconnected peripheral as well. I believe that rounds this out enough.